public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -16,6 +16,66 @@ export function alignRange(start, end, step) {
   };
 }
 
+const keywords = 'by|without|on|ignoring|group_left|group_right';
```

Comment:
Could we reuse it between `mode-prometheus.js` and this file? In future, it may help us to migrate to Slate.